### DISCUSSION TITLE
Default Search Plugin for Pimcore X/11?

### DISCUSSION THREAD

**Discussion by beheist** [pimcore/pimcore#15997] [Q&A / Support]
Hey all,

I am wondering: What kind of search module are you using with Pimcore 11? What do you recommend as the default?
With Pimcore 6, we have been using [Lucene Search](https://github.com/dachcom-digital/pimcore-lucene-search) . With some monkeypatching, I made this work with Pimcore X as well, since the new Dynamic Search was not ready yet.

With Pimcore 11, we have tried once again to get dynamic search running. I still cannot figure out how to make it work, since many of the concepts mentioned in the docs are not explained and there is no documentation on how to actually get it to index your site. It seems that you need some sort of document definition, but the relevant parts of the documentation are missing.
How do I get a working search in Pimcore 11? Starting to fear I have to monkeypatch lucene search once again.

Thanks for your feedback.
cc @DGrosmayer

**Comment by Ralff212** [comment]
Have you found a solution for this? I don't have a "search" menu or icon either...

**Comment by aarongerig** [comment]
@beheist @Ralff212 I would highly suggest to use [Dynamic Search](https://github.com/dachcom-digital/pimcore-dynamic-search). It's the best OS frontend search bundle there is for Pimcore.

How indexing works is documented as well [here](https://github.com/dachcom-digital/pimcore-dynamic-search/blob/master/docs/0_ExampleSetup.md#indexing). Also, I would recommend you to use it with the [Trinity Data Provider](https://github.com/dachcom-digital/pimcore-dynamic-search-data-provider-trinity). With that it will index your DataObjects/Documents on save, so changes are present in your search results immediately.

Let me know if you have more questions!

**Reply by DGrosmayer** [reply]
Hi @aarongerig, I've tried the example documted in your link with trinity provider + lucene indexer but I cant seem to get any results if i call "/dynamic-search/default/j-search?q=xyz" (and xyz is any content i have put on my pages) as the array is empty. The "bin/console dynamic-search:run -v" command gave me some feedback about adding e.g. documents to the index though. Do you maybe know about an example I can look over which is more recent?  I've seen the MultiSite Setup example, but that one runs with elastic which I want to avoid.
